### DISCUSSION TITLE
Improved smartlog clean up

### DISCUSSION THREAD

**Discussion by pink** [arxanas/git-branchless#977] [Q&A]
Has there been any discussions yet around workflows or strategies to keep the smartlog clean?

With my current workflow, I'll often have various branches in my local checkout that were either abandoned without merging, or merged into `master`. For the former, I think the current solution is to `git hide` the branches/commits after I know I'm done with them. For the latter, `git sync` sometimes works, but my repos enforce a squash-and-merge commit style, so usually the merged branch will get passed over due to "merge conflicts".

I was thinking about writing a utility to make bulk clean up a bit easier, but I'm wondering if there were any other tips around managing the mess.

**Comment by arxanas** [comment]
My recommendation in https://github.com/arxanas/git-branchless/discussions/965 is to squash the commits manually before merging on GitHub. For general cleanliness, you could also change your default smartlog revset to intersect with something like `committer.date('after: 2 weeks ago')` to omit branches which are likely to be out-of-date.
